07.bowling_object/frame.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class Frame
+  attr_reader :frame_score
+
+  def initialize(first_mark, second_mark, third_mark = nil)
+    @first_shot = Shot.new(first_mark)
+    @second_shot = Shot.new(second_mark)
+    @third_shot = Shot.new(third_mark)
+    @frame_score = calc_frame_score
+  end
+
+  private attr_reader :first_shot, :second_shot, :third_shot # rubocop:disable Style/AccessModifierDeclarations
+
+  def strike_or_spare?
+    first_shot.score == 10 || [first_shot, second_shot].map(&:score).sum == 10
+  end
```

Comment:
`Game`クラスにも`strike?`メソッドがありますね。あるフレームがストライクかどうかを判断する知識が`Frame`クラスと`Game`クラスの両方に散らばってしまっている（しかも微妙にロジックが違う）ので、バグの原因になりそうです。
「あるフレームがストライクかどうかをどうやって判定するか」という知識を知っているのはただ一つのクラスであるべきなので、`Frame`か`Game`かどちらかに寄せましょう。（個人的には`Frame`がいいと思いますが、`Game`でも良さそうです）

ついでにですが、ロジックの重複を防ぐために、`spare?`も定義して↓のような感じにした方が良さそうかなと思います。（あくまでイメージです）

```suggestion
  def strike_or_spare?
    strike? || spare?
  end
```

…というか、ここまできたらもはやこのメソッドは必要なさそうですね。。最終的にどのような形にするのがいいかは考えてみてください〜

Reply:
ありがとうございます！

`strike?`周辺の設計はかなりモヤモヤしながら提出したのでアドバイス助かりました！

今回は`Frame#strike?`をpublicメソッドに定義し、`Game#create_frames`内でも呼び出せるようにしました。記述も減り、可読性も上がったように思います😁